src/Streamly/Internal/Data/Stream/IsStream/Top.hs

```diff
@@ -391,13 +391,26 @@ leftJoin eq s1 s2 = Stream.evalStateT (return False) $ do
 --
 -- Time: O(m + n)
 --
--- /Unimplemented/
-{-# INLINE hashLeftJoin #-}
-hashLeftJoin :: -- Hashable b =>
-    (a -> b -> Bool) -> t m a -> t m b -> t m (a, Maybe b)
-hashLeftJoin = undefined
+-- /Pre-release/
+{-# INLINE joinLeftMap #-}
+joinLeftMap :: (IsStream t, Ord k, Monad m) =>
+    t m (k, a) -> t m (k, b) -> t m (k, a, Maybe b)
+joinLeftMap s1 s2 =
+    Stream.concatM $ do
+        km <- kvFold $ IsStream.adapt s2
+        return $ Stream.map (joinAB km) s1
+
+            where
+
+            -- XXX Generate error if a duplicate insertion is attempted?
+            kvFold = Stream.foldl' (\kv (k, b) -> Map.insert k b kv) Map.empty
```

Comment:
We can take this out at top level and name it `toMap`, so that it can be reused across functions.